File: sbschoolmanager-api/src/main/java/com/sbschoolmanager/sbschoolmanager_api/model/Alumno.java
package com.sbschoolmanager.sbschoolmanager_api.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;

@Entity
@Table(name="ALUMNO")
@Getter
@Setter
@NoArgsConstructor
public class Alumno {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name= "Cod_alumno")
    private Integer codAlumno;

    @Column(name="Nom_alumno_CC")
    private String nombre;

    @Column(name="Ap_alumno")
    private String apellido;

    @Column(name="DNI_alumno")
    private String dni;

    @Column(name = "Edad")
    private Integer edad;

    @Column(name = "Datos_Padres_OP")
    private String datosPadres;
}

File: sbschoolmanager_app/lib/models/alumno.dart

class Alumno {
  final int? codAlumno; //El id lo asigna la BD
  final String nombre;
  final String apellido;
  final String dni;
  final int edad;
  final String? datosPadres;

  Alumno({
      this.codAlumno,
      required this.nombre,
      required this.apellido,
      required this.dni,
      required this.edad,
      this.datosPadres,
  });

  //Json de Spring -> objeto Alumno
  factory Alumno.fromJson(Map<String, dynamic> json){
    return Alumno(
      codAlumno: json["codAlumno"],
      nombre: json["nombre"], 
      apellido: json["apellido"],
      dni: json["dni"], 
      edad: json["edad"],
      datosPadres: json["datosPadres"],
      );
  }

  //Objeto Alumno -> JSON
  Map<String, dynamic> toJson(){
    return{
      if(codAlumno != null)
      "codAlumno": codAlumno,
      "nombre": nombre,
      "apellido": apellido,
      "dni": dni,
      "edad": edad,
      "datosPadres": datosPadres,
    };
  }
}